### PR TITLE
Look for matching Conan profile in a config file for the running VS configuration

### LINKED ISSUE #88: Support mapping of configuration names -> build_type
Currently, the extension uses the configuration name as the build_type, and there is no way to override this.  This becomes an issue when a project has multiple custom configurations which, from Conan's perspective, all need to map to the same standard "build_type" when installing dependencies.  

E.g. an existing project has a "Release" configuration and a "Native_Windows_Release" configuration.  It should be possible to specify that the extension should treat "Native_Windows_Release" as just "Release" when installing for the project, as it currently does not work otherwise.

### PR DESCRIPTION
**Pull request by jgsogo**
The extension looks into a file `conan.config.json` for a Conan profile name associated with the Visual Studio configuration. This file is searched first in the same directory as the VS Project, and then in any parent directory.

The format for this file (it should be extensible in the future):

```json
{  
   "configurations": {  
      "Release|Win32": "release_win32",
      "Debug|x64": "debug_x64",
      "Release|Win64": "C:/conan/profiles/release_win64"
   }
}
```

Under the `"configurations"` key there is a dictionary with the correspondence between the VS configuration (key, on the left) and a Conan profile (value, on the right). To take into account:
 * VS creates the configuration name joining with a `|` character the name of the `Configuration` and the `Platform`.
image
 * For Conan profiles, the value declared will be used verbatim for the `--profile` argument in the `conan install` command, and rules related to profile lookup applies.

closes #88
related to #5 (first iteration)

### CODE REVIEW

File: Conan.VisualStudio.Core/ConanProject.cs [hunk 1]
```diff
@@ -1,11 +1,31 @@
...
+
+        public string getProfile(ConanConfiguration configuration)
+        {
+            if (!File.Exists(ConfigFile)) return null;
+            try
+            {
+                JObject jObject = JObject.Parse(File.ReadAllText(ConfigFile));
+                var configs = jObject["configurations"].ToObject<Dictionary<string, string>>();
+                configs.TryGetValue(configuration.VSName, out string conanProfile);
+                return conanProfile;
```
Comment (jgsogo) [inline Conan.VisualStudio.Core/ConanProject.cs]: We should write a warning message (_"File `conan.config.json` file available for the project, but no profile declared for VS configuration 'xxxx'"_), but I'm not sure how are we handling the output. @SSE4, can you point me in the right direction?

File: Conan.VisualStudio.Core/ConanProject.cs [hunk 2]
```diff
@@ -1,11 +1,31 @@
...
+            }
+            catch (Exception)
+            {
+                // TODO: Check the error and log to user
+                return null;
```
Comment (jgsogo) [inline Conan.VisualStudio.Core/ConanProject.cs]: Same here, log to output _"Error reading conan.config file: 'xxxxxx'"_. @SSE4 , how can I do it?

File: Conan.VisualStudio.Core/ConanProject.cs (hunk 1 above)
Comment (SSE4) [reply Conan.VisualStudio.Core/ConanProject.cs]: we have two ways to output messages.
1. `ErrorList` Window - use `ErrorListService` for this
2. `Output` Window - use `Logger` class
I think for this one it's better to use ErrorList, and double click on error may open `conan.config.json` with incorrect line

File: Conan.VisualStudio.Core/ConanRunner.cs [hunk 3]
```diff
@@ -24,7 +24,17 @@ public Task<Process> Install(ConanProject project, ConanConfiguration configurat
...
-            if (_conanSettings != null)
+
+            string profile = project.getProfile(configuration);
+            if (profile != null)
+            {
+                string generatorName = generator.ToString();
+                arguments = $"install {Escape(project.Path)} " +
+                            $"-g {generatorName} " +
+                            $"--install-folder {Escape(configuration.InstallPath)} " +
+                            $"--profile {Escape(profile)}";
```
Comment (SSE4) [inline Conan.VisualStudio.Core/ConanRunner.cs]: it ignores `--build` and `--update` arguments, which are currently set in global settings

File: Conan.VisualStudio.Core/ConanProject.cs (hunk 1 above)
Comment (jgsogo) [reply Conan.VisualStudio.Core/ConanProject.cs]: But I don't have the `ErrorListService` in the `Core` project...

Comment (SSE4) [reply Conan.VisualStudio.Core/ConanProject.cs]: then it needs to get delegate/callback function for error reporting

File: Conan.VisualStudio.Core/ConanRunner.cs (hunk 3 above)
Comment (jgsogo) [reply Conan.VisualStudio.Core/ConanRunner.cs]: Done!

File: Conan.VisualStudio.Core/ConanProject.cs (hunk 1 above)
Comment (jgsogo) [reply Conan.VisualStudio.Core/ConanProject.cs]: probably the _Interface_ in the `Core` and the implementation in the other project, I've read more about this pattern than using _callbacks_, right?

Comment (SSE4) [reply Conan.VisualStudio.Core/ConanProject.cs]: something like `IAbstractErrorReporting` interface which might be passed to the path helper. then `ErrorListService`, `DialogService` or `OutputterService` (class to write into output window) may inherit from this interface, and could be passed into the core.

Comment (jgsogo) [reply Conan.VisualStudio.Core/ConanProject.cs]: Done 👍

**Comment by jgsogo** [comment]
Please, @SSE4 , review again and test the functionality. I think it could be almost ready. 😸

**Comment by SSE4** [comment]
I think it deserves an example project with:
1. custom configuration names
2. json config file
3. profiles for it

**Comment by jgsogo** [comment]
Very basic example added, check it works as expected (using the `visual_studio` generator #108). Thanks

**Comment by SSE4** [comment]
works for me, please un-draft the PR

**Pull request merged**